### DISCUSSION TITLE
Share Qt Designer Custom Widget plugins

### DISCUSSION THREAD

**Discussion by Prx001** [MadPonyInteractive/MadQt#5] [Q&A] [answered]
How can I share Qt Designer Custom Widget plugins?

**Comment by MadPonyInteractive** [comment, marked as answer]
Hi Prx001,
Glad to know you are using the MadQt package.
There's a few ways to do it.

- The manual way:
You would need to share the produced files and the end user would have to point the environment variable PYSIDE_DESIGNER_PLUGINS to the files directory.

- The installer way:
You could create an installer that automatically does the above
Here's one for windows: https://jrsoftware.org/isinfo.php

To automatically add environment paths I've been using a couple of functions that you can find in the __ init __.py file of the MadQt package, setPermanentEnv() and addEnvPath()

You could also add them to MadQt, I will be adding some soon so they can be used as an example.

I've been working on improving Plugin Creator and will have some new features on upcoming versions, and also will consist of less files.

Hope that helps.

**Comment by Prx001** [comment]
Thanks,
but how can I add them to MadQt to make 'em popular?

**Comment by MadPonyInteractive** [comment]
At the moment there are no plugins setup in MadQt yet so it's hard to figure out how but I will add one in version v0.0.29 so you and others can see how it is setup and contribute that way.

If you are new to contributing here's some videos to get you started: https://www.youtube.com/results?search_query=github+contributing

The objective at the moment is to add your widget to the project Widgets folder and your plugin to the QtDesignerPlugins folder but once I add the first one it will be clear.

Thank you so much for your interest in being part of this project.

I will revisit and comment in this post once v0.0.29 is released.

**Comment by MadPonyInteractive** [comment]
> Thanks, but how can I add them to MadQt to make 'em popular?

I just updated MadQt to v0.0.29 and added a plugin there to serve as example.

Also added a tutorial in the documentation: https://madponyinteractive.github.io/MadQt/add_plugins.html
